src/main/kotlin/app/UserService.kt

```diff
@@ -97,18 +66,36 @@ object UserService {
         listOf()
     }
 
-}
+    private fun generateUserProfile(username: String): UserProfile {
```

Comment:
good refactor here 👍

Reply:
Nice. Let me know if I got it wrong.
https://github.com/Attacktive/profile-summary-for-github/commit/8306d9210135a2ad5d876d22e4a51e43626a8b4a

Reply:
thanks for giving this a shot - using the `username` directly unfortunately won't do the job (synchronizing on `username.intern()` *would* work but isn't the most efficient solution). I'm imagining a `ConcurrentMap<String, Lock>` but is slightly tricky so we can leave this for a separate PR (& we can revert https://github.com/Attacktive/profile-summary-for-github/commit/8306d9210135a2ad5d876d22e4a51e43626a8b4a for now)